src/picongpu/include/particles/manipulators/RatioWeightingImpl.hpp

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright 2015 Axel Huebl
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+#pragma once
+
+#include "particles/manipulators/RatioWeightingImpl.def"
+#include "particles/traits/GetDensityRatio.hpp"
+
+#include "simulation_defines.hpp"
+
+namespace picongpu
+{
+namespace particles
+{
+namespace manipulators
+{
+
+struct RatioWeightingImpl
+{
+
+    template<typename T_SpeciesType>
+    struct apply
+    {
+        typedef RatioWeightingImpl type;
+    };
+
+    HINLINE RatioWeightingImpl(uint32_t currentStep)
```

Comment:
since `currentStep` is not needed in the `struct` could you avoid the name and define the function as 
```
HINLINE ProtonTimesWeightingImpl(uint32_t ) 
```